test/Common.FunctionalTests/OutOfProcess/AspNetCorePortTests.cs

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Net.Sockets;
+using System.Threading.Tasks;
+using Xunit;
+using Microsoft.AspNetCore.Server.IIS.FunctionalTests.Utilities;
+using Microsoft.AspNetCore.Server.IntegrationTesting;
+using Microsoft.AspNetCore.Testing.xunit;
+
+namespace Microsoft.AspNetCore.Server.IISIntegration.FunctionalTests
+{
+    [Collection(PublishedSitesCollection.Name)]
+    public class AspNetCorePortTests : IISFunctionalTestBase
+    {
+        private static readonly Random _random = new Random();
+
+        private readonly PublishedSitesFixture _fixture;
+
+        public AspNetCorePortTests(PublishedSitesFixture fixture)
+        {
+            _fixture = fixture;
+        }
+
+        public static TestMatrix TestVariants
+            => TestMatrix.ForServers(DeployerSelector.ServerType)
+                .WithTfms(Tfm.NetCoreApp22)
+                .WithApplicationTypes(ApplicationType.Portable)
+                .WithAllAncmVersions();
+
+        [ConditionalTheory]
+        [MemberData(nameof(TestVariants))]
+        public async Task EnvVarInWebConfig(TestVariant variant)
+        {
+            // Must publish to set env vars in web.config
+            var deploymentParameters = _fixture.GetBaseDeploymentParameters(variant, publish: true);
+
+            // Workaround for GetBaseDeploymentParameters hardcoding ANCM v2
+            deploymentParameters.AncmVersion = variant.AncmVersion;
+
+            var port = GetUnusedRandomPort().ToString();
+            deploymentParameters.WebConfigBasedEnvironmentVariables["ASPNETCORE_PORT"] = port;
+
+            var deploymentResult = await DeployAsync(deploymentParameters);
+
+            var response = await deploymentResult.HttpClient.GetAsync("/AspNetCorePortEnvVar");
```

Comment:
I would prefer asserting that we are actually listening on the specified port.

Reply:
OK, this should work for outofprocess, but there's currently an issue claiming it doesn't work for inprocess: https://github.com/aspnet/IISIntegration/issues/1223

Reply:
We don't care about inprocess in this PR.